documentation/src/main/asciidoc/reference/search-dsl-paths.asciidoc

```diff
@@ -0,0 +1,82 @@
+[[search-dsl-paths]]
+= Field paths
+
+[[search-dsl-paths-absolute]]
+== Absolute field paths
+
+By default, field paths passed to the Search DSL are interpreted as absolute,
+i.e. relative to the index root.
+
+The components of the paths are separated by a dot (`.`).
+
+[NOTE]
+====
+The following examples use the <<search-dsl-predicate,predicate DSL>>,
+but all information in this section applies to other search DSLs as well:
+<<search-dsl-sort,sort DSL>>, <<search-dsl-projection,projection DSL>>, <<search-dsl-aggregation,aggregation DSL>>, ...
+====
+
+.Targeting a field using absolute paths
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/paths/FieldPathsIT.java[tags=root]
+----
+<1> Fields declared at the root of the index can simply be referenced by their name.
+
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/paths/FieldPathsIT.java[tags=nested_implicit]
+----
+<1> Fields declared in object fields (created by <<mapper-orm-indexedembedded,`@IndexedEmbedded`>>, for example)
+must be referenced by their absolute path: the absolute path of the object field, followed by a dot,
+followed by the name of the targeted field.
+
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/paths/FieldPathsIT.java[tags=nested_explicit]
+----
+<1> Even within a `nested` predicate, fields referenced in inner predicates must be referenced by their absolute path.
```

Comment:
That would be a backward-incompatible change. Not going to happen in 6.x.

At best we can provide alternative methods to that end, but even then it will be challenging to avoid confusion.

Most examples use a single factory `f` obtained from the lambda passed to `.where( f -> ... )`, and that factory obviously uses absolute paths. In order to allow relative paths, we would have to pass a new factory to the user, e.g. `.where( f -> f.nested().objectField( "myObject", f2 -> ... )`. While it works, it quickly becomes unmanageable, and you quickly end up mixing `f` with `f2`, which is a problem since they don't have the same root.

In fact, this is exactly how Hibernate Search used to work at some point while we were still prototyping it, but we just gave up on this syntax because it was too confusing.

Feel free to create a ticket, but keep in mind that while the implementation part would now be easy, the API part is very challenging.